tornjak-backend/api/agent/server.go

```diff
@@ -527,15 +553,74 @@ func (s *Server) HandleRequests() {
 }
 
 // NewAgentsDB returns a new agents DB, given a DB connection string
-func NewAgentsDB(dbString string) (agentdb.AgentDB, error) {
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.MaxElapsedTime = time.Second
+func NewAgentsDB(dbPlugin map[string]catalog.HCLPluginConfig) (agentdb.AgentDB, error) {
+	//dbString = "./agentlocaldb"
+	key := ""
+	var data ast.Node
+	for k, d := range dbPlugin {
+		key = k
+		data = d.PluginData
+	}
+
+	if key == "" { // error if no config
+		return nil, errors.Errorf("No Datastore plugin config provided")
+	} else if key == "sql" {
+		expBackoff := backoff.NewExponentialBackOff()
+		expBackoff.MaxElapsedTime = time.Second
+
+		var config map[string]interface{}
+		if err := hcl.DecodeObject(&config, data); err != nil {
+			return nil, errors.Errorf("Couldn't parse DB config: %v", err)
+		}
+		drivername := config["databaseDrivername"].(string)
+		dbfile := config["dbString"].(string)
+
+		db, err := agentdb.NewLocalSqliteDB(drivername, dbfile, expBackoff)
+		if err != nil {
+			return nil, err
+		}
+		return db, nil
+	}
+	
+	return nil, errors.Errorf("Couldn't create datastore")
+}
 
-	db, err := agentdb.NewLocalSqliteDB(dbString, expBackoff)
+// NewAuth returns a new Auth
+func NewAuth(authPlugin map[string]catalog.HCLPluginConfig) (auth.Auth, error) {
+	key := ""
+	var data ast.Node
+	for k, d := range authPlugin {
+		key = k
+		data = d.PluginData
+	}
+	if key == "" { // default is null verifier
+		verifier := auth.NewNullVerifier()
+		return verifier, nil
+	} else if key == "KeycloakAuth" {
+		var config map[string]interface{}
```

Comment:
this should be a struct instead , would make it clearer what is expected and how the `auth` package would use the config.. It would be helpful in general to have some documentation for configuration and a sample config file somewhere